daemon/cmd/daemon.go

```diff
@@ -667,7 +667,7 @@ func NewDaemon(ctx context.Context, cancel context.CancelFunc, epMgr *endpointma
 	// FIXME: Make the port range configurable.
 	if option.Config.EnableL7Proxy {
 		d.l7Proxy = proxy.StartProxySupport(10000, 20000, option.Config.RunDir,
-			&d, option.Config.AgentLabels, d.datapath, d.endpointManager)
+			&d, option.Config.AgentLabels, d.datapath, d.endpointManager, ipcache.IPIdentityCache)
```

Comment:
Are we using a global var here? `ipcache.IPIdentityCache` ? Can be OK just a little confused since commit seems to want to remove global usages.

Reply:
Overall the PR first removes global usages from individual packages under `pkg/` commit by commit, which pushes the reference up to the daemon. That way the individual commits can be constrained mostly to only changing a couple of files focused on one area of the codebase. Then later on, one of the patches goes through all of these references in the daemon package only and makes sure they're all referring to the canonical one that the daemon initializes. It looks like you caught this while stepping through commit-by-commit, as the version on the tip of the tree now refers to `d.ipcache` (and also `ipcache.IPIdentityCache` variable is removed, so the build would fail if it was still there at the end of the whole PR).